src/compiler/utilities.ts

```diff
@@ -1730,6 +1730,14 @@ namespace ts {
         return !!node && isVariableDeclaration(node) && node.initializer?.kind === SyntaxKind.ThisKeyword;
     }
 
+    export function isThisInitializedExpression(node: Node | undefined): boolean {

```

Comment:
`isThisInitializedObjectBindingExpression`? (I'm not sure that ObjectBindingExpression is what we call destructuring assignment inside the compiler)